### DISCUSSION TITLE
Apple Pay with Stripe in Checkout

### DISCUSSION THREAD

**Discussion by edgarlr** [swellstores/community#211] [Help] [answered]
I have a custom storefront and use Swell as with the hosted checkout. I already integrated Stripe as a Payment getaway and enabled Apple Pay on both sides ad the documentation specify, but now I'm getting this error on the console of the checkout:

`
We were unable to verify that you control the domain moms-farm-to-table.swell.store. When we tried to request https://moms-farm-to-table.swell.store/.well-known/apple-developer-merchantid-domain-association, it redirected to https://moms-farm-to-table.swell.store/admin. The verification file must be served as a 200 at https://moms-farm-to-table.swell.store/.well-known/apple-developer-merchantid-domain-association to register moms-farm-to-table.swell.store. Did you mean to register moms-farm-to-table.swell.store? For more information, see https://stripe.com/docs/stripe-js/elements/payment-request-button#verifying-your-domain-with-apple-pay.|
`

if I hit: `https://moms-farm-to-table.swell.store/.well-known/apple-developer-merchantid-domain-association` it does redirect to `/admin` but I'm not sure how to turn off the redirect or know if there's some missing configuration to load the Apple Pay in the store since in both Swell and Stripe says is properly connected.

Does anyone know how to either turn off the redirect on the apple pay script or if there's some additional setup step I might be missing?

**Comment by sags95** [comment]
@edgarlr Do you see any domains registered in https://dashboard.stripe.com/settings/payments/apple_pay

**Comment by mewa** [comment, marked as answer]
You need to include the verification file.

It also was a bit confusing for us at first, so we documented the whole process here: https://swelltoolkit.com/blog/swell-enable-apple-pay-with-stripe

**Reply by sags95** [reply]
We're currently investigating an issue where domains are not configured automatically during setup

**Reply by edgarlr** [reply]
@sags95 any updates on this? Seems that @mewa has the right answer but since I'm still using the hosted checkout and with all the recent changes on Swell, I haven't been able to either add the static file (apple verification file) within Swell dashboard or set a custom domain for my checkout so I can self-host the static file there.

**Reply by sags95** [reply]
@edgarlr We're actively working on a fix for this, I should have more details on ETA as it moves through our process.

**Reply by edgarlr** [reply]
@sags95 great,thanks! and in the meantime, do you know if I can either ser a custom domain for my swell-hosted checkout (I tried but the domain tab is no visible anymore in my dashboard) to host the file myself or use git ssh to push the file also myself as @mewa suggested (the ssh tab is also not available anymore in my dashboard) I've been blocked for the apple pay support in the checkout a couple of weeks now and I do need to close the project as soon as posible, any alternative or workaround will be great, thanks.

**Reply by steven-k1** [reply]
A fix has been deployed for the raised issue with store URLs not being added to the Apple Pay domain list in Stripe.

**Comment by logeshswell** [comment]
@edgarlr In the meantime of rolling out the fix, I will help you with adding the custom domain for your Swell-hosted checkout as a temporary fix.
As the first step, you need to create CNAME record with your custom domain pointing to `moms-farm-to-table.apps.swellstores.com`
You can ping me if the configuration is done and the CNAME verification is successful from the available [CNAME lookup](https://dnschecker.org/cname-lookup.php?query=momsfarmtotable.com&dns=cloudflare) sites.

**Reply by edgarlr** [reply]
@logeshswell Awesome, thanks! the CNAME is now set. Since the website is under the root domain, the store and checkout are intended to be under `store.momsfarmtotable.com` the [CNAME lookup](https://dnschecker.org/cname-lookup.php?query=store.momsfarmtotable.com&dns=cloudflare) correctly shows is pointing to the correct domain

Please let me know how it goes and if there's any other change required, thanks again!

**Reply by logeshswell** [reply]
@edgarlr As per your request, the swell-hosted checkout will use your custom domain, and here is the sample checkout URL:
https://store.momsfarmtotable.com/checkout/c9c97ce6aeae80d2f681ad3975c83bd5

**Reply by fyodorp** [reply]
@logeshswell are you guys still planning to keep the hosted checkout, or there will be nothing swell-hosted anymore?

**Reply by ericingram** [reply]
We will definitely keep supporting hosted checkout. There are other good options as well, like Rally or customs

**Reply by fyodorp** [reply]
@ericingram @logeshswell Due to all the recent changes, I feel it would be more efficient for me to only stick with the hosted checkout from Swell and use checkout links.
However, with domain mapping being deprecated, I need some clarification. Is there still a way for me to map the domain name on my own, or will I need to reach out for support, similarly to a topic starter?
Please point me in the right direction.

**Reply by ericingram** [reply]
It is still possible, and we’re working on a way to make to more obvious

**Reply by fyodorp** [reply]
@ericingram Could you please provide more details or perhaps step-by-step instructions about such a procedure?

**Reply by ericingram** [reply]
@StevenSwell can you share a link to a docs page on this topic - if we don't have one yet, let's create one

**Reply by logeshswell** [reply]
> However, with domain mapping being deprecated, I need some clarification. Is there still a way for me to map the domain name on my own, or will I need to reach out for support, similarly to a topic starter?

@fyodorp We are working towards making it very easy to configure the custom domains specific to checkout.

In the meantime, if your store has at least one swell-hosted storefront, then still you could configure the custom domain on your own under the **Domains** tab in Storefront settings. Otherwise, you would need the assistance of Swell support.

With the current implementation, if you configured the custom domain for your swell-hosted storefront, then it would be applicable for swell-hosted checkout as well.

Can you share me your `store-id` so that I can check and confirm you?

**Comment by fyodorp** [comment]
It is easebee

**Reply by logeshswell** [reply]
@fyodorp This store has a origin storefront which is already mapped to `www.easebee.com`. You would also see `checkout_url` with this domain. Do you wish to change it to a new domain?

**Comment by fyodorp** [comment]
I see that, thanks!
Is there a way though to only map the checkout, but not the website?

**Reply by logeshswell** [reply]
No, I don't think so
